lib/updatehub/updatehub.c

```diff
@@ -703,9 +705,18 @@ enum updatehub_response updatehub_probe(void)
 			goto cleanup;
 		}
 
+		sha256size = strlen(
+			metadata_any_boards.objects[1].objects.sha256sum);
+
+		if (sha256size != TC_SHA256_BLOCK_SIZE) {
+			LOG_ERR("SHA256 size is invalid");
+			ctx.code_status = UPDATEHUB_METADATA_ERROR;
+			goto cleanup;
+		}
+
 		memcpy(update_info.sha256sum_image,
 		       metadata_any_boards.objects[1].objects.sha256sum,
-		       strlen(metadata_any_boards.objects[1].objects.sha256sum));
+		       TC_SHA256_BLOCK_SIZE);
```

Comment:
Do we still need to make sure there is a terminating null on the string? It does seem to be used as a string elsewhere in the code.

Reply:
Given that you've already verified the length, maybe the best thing to do here is actually just call 'strcpy'. There is a 'strlcpy' in newlib, but it doesn't seem to be enabled in our configuration. But, if that would happen, that would be the cleanest way to do all of this.
But, a memcpy should be fine, as long as you also set the null termination at the end.

Reply:
Now we use total hex string size on memcpy operations.